### ISSUE TITLE
Retires persist forever if blocked by adblocker

### ISSUE THREAD

**Issue by chasegiunta** [PostHog/posthog-js#1598]
I'm aware that using the reverse proxy might be a workaround to this, but at the same time, I think this should be addressed. I'd think this would be a configurable option in the client. Also, might be related, it looks like the retry param isn't being updated in these failed requests?

<img width="1393" alt="Firefox 2024-12-11 12 51 32" src="https://github.com/user-attachments/assets/d8be3f44-4479-4656-a3ed-cab064fa9ef2" />

**Comment by rafaeelaudibert** [comment]
Hey, @chasegiunta! Are you sure `posthog-js` is retrying always the same request? Are the payloads exactly the same?

Also, would you mind calling `posthog.debug()` and showing me what's being logged to the console?

I can help you with that if you could share me a URL where I can see that behavior myself?

**Comment by chasegiunta** [comment]
Hey @rafaeelaudibert, thanks for getting back to me so quick! Unfortunately, we had to remove the frontend client from our live application for now.

Here's debug output
<img width="1886" alt="Firefox 2024-12-11 13 21 31" src="https://github.com/user-attachments/assets/bc6ebdf6-2e39-4ce6-a23c-63492179a0fa" />

And here are some more requests. 
<img width="1888" alt="Firefox 2024-12-11 13 26 52" src="https://github.com/user-attachments/assets/64fef6b0-546b-490c-8d89-eab5133f5512" />

I believe there's a few initial ones that are blocked, which seem fine, but the others that are repetitive at the bottom, all seem to be exactly the same. (all with `retry_count=1` params) Firefox does not actually show a payload.

<img width="639" alt="image" src="https://github.com/user-attachments/assets/2a13b9cc-3218-47ad-a367-24ad04fe133a" />

<img width="1076" alt="image" src="https://github.com/user-attachments/assets/f28329c0-9b9a-422f-a8df-960700ee7076" />

**Comment by rafaeelaudibert** [comment]
Yeah, it's not displaying the payload because this is the initial HEAD request, it's failing before we can actually POST  the data. We'll investigate further, but I can't promise on a deadline, thank you!

**Comment by Julien-Marcou** [comment]
We had the same issue about a year ago, I think this ticket is talking about the same issue: https://github.com/PostHog/posthog-js/issues/482
I guess it's something that could also happens if posthog's servers are down for whatever reason.

PostHog being blocked by an adblocker is not necessary a problem, until we want to do A-B testing on our entire platform using feature flags.

Because of the adblocker, the `onFeatureFlags()` callback would never be called, which means we were leaving our users on an infinite loading blank screen.
So we had to add a global timeout, to fallback to a default feature flag value to resolve blank screen issue.

It would be nice to have this behavior configurable, with something like a "max retry" or "timeout", and an error callback on the `onFeatureFlags` method:

```ts
// example
import posthog from 'posthog-js';

posthog.onFeatureFlags({
  loaded: (flags, variants) => {
    // feature flags are properly loaded
  },
  error: () => {
   // fallback when we cannot load the feature flags
  },
  retries: 3,
});

```

**Comment by Predhin** [comment]
+ 1

Hi folks, can we have the retries configurable? Thanks
